server/routes/referrals/referralDataPresenterUtils.test.ts

```diff
@@ -0,0 +1,252 @@
+          const userInputData = {}
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBe(null)
+        })
+      })
+
+      describe("when there is user input data and the user input value for the property is 'no'", () => {
+        it('returns false', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = { 'needs-interpreter': 'no' }
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBe(false)
+        })
+      })
+
+      describe("when there is user input data and the user input value for the property is 'yes'", () => {
+        it('returns true', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = { 'needs-interpreter': 'yes' }
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBe(true)
+        })
+      })
+
+      describe("when there is user input data and the user input value for the property is present but neither 'yes', 'no' nor null", () => {
+        it('returns null', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: null })
+          const userInputData = { 'needs-interpreter': true }
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBeNull()
+        })
+      })
+    })
+
+    describe('when the referral has a non-null value for the property', () => {
+      describe('when there is no user input data', () => {
+        it('returns the value from the referral', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: false })
+          const utils = new ReferralDataPresenterUtils(referral, null)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBe(false)
+        })
+      })
+
+      describe('when there is user input data and the user input value for the property is null', () => {
+        it('returns null', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: false })
+          const userInputData = { 'needs-interpreter': null }
+          const utils = new ReferralDataPresenterUtils(referral, userInputData)
+
+          expect(utils.booleanValue('needsInterpreter', 'needs-interpreter')).toBeNull()
+        })
+      })
+
+      describe('when there is user input data and the property is absent from the user input data', () => {
+        it('returns null', () => {
+          const referral = draftReferralFactory.build({ needsInterpreter: false })
+          const userInputData = {}
```

Comment:
same again, i'm slightly confused by this one.

Reply:
Yeah, I think same answer as my previous one.